### PR TITLE
Instrument ConfigAccessor's constructors

### LINKED ISSUE #839: Make ConfigAccessor and ZkUtil realm-aware
To make Helix Java APIs realm-aware, we first make ConfigAccessor and ZkUtil realm-aware by instrumenting these APIs with a Builder and RealmAwareZkClients.

### PR DESCRIPTION
**Pull request by narendly**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

Fixes #839 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This diff instruments ConfigAccessor's constructors to make it realm-aware. If ConfigAccessor is unable to start on multi-realm mode, then it falls back to starting on single-realm mode.

### Tests

- [x] The following tests are written for this issue:

TestConfigAccessor

- [x] The following is the result of the "mvn test" command on the appropriate module:

**helix-core:**

[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TestConfigAccessor.testBasic:46 » IllegalState HttpRoutingDataReader was unabl...
[ERROR]   TestEnableCompression.testEnableCompressionResource:108 expected:<true> but was:<false>
[ERROR]   TestControllerLeadershipChange.testMissingTopStateDurationMonitoring:262 expected:<true> but was:<false>
[INFO] 
[ERROR] Tests run: 1083, Failures: 3, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:07 h
[INFO] Finished at: 2020-03-04T11:42:53-08:00
[INFO] -----------------------------------------

All tests pass when run individually.

### Commits

- [x] My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation (Optional)

- [ ] In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Code Quality

- [x] My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-core/src/main/java/org/apache/helix/ConfigAccessor.java [hunk 1]
```diff
@@ -111,7 +111,7 @@ private ConfigAccessor(Builder builder) throws IOException, InvalidRoutingDataEx
    * @param zkClient
```
Comment (pkuwm) [inline helix-core/src/main/java/org/apache/helix/ConfigAccessor.java]: Nit: change the java doc: "HelixZkClient" -> "RealmAwareZkClient"

File: helix-core/src/main/java/org/apache/helix/ConfigAccessor.java [hunk 2]
```diff
@@ -123,9 +123,22 @@ public ConfigAccessor(HelixZkClient zkClient) {
...
-    _zkClient = SharedZkClientFactory.getInstance()
-        .buildZkClient(new HelixZkClient.ZkConnectionConfig(zkAddress),
-            new HelixZkClient.ZkClientConfig().setZkSerializer(new ZNRecordSerializer()));
+    // First, attempt to connect on multi-realm mode using FederatedZkClient
+    RealmAwareZkClient zkClient;
+    try {
+      zkClient = new FederatedZkClient(
+          new RealmAwareZkClient.RealmAwareZkConnectionConfig.Builder().build(),
+          new RealmAwareZkClient.RealmAwareZkClientConfig());
+    } catch (IOException | InvalidRoutingDataException e) {
```
Comment (pkuwm) [inline helix-core/src/main/java/org/apache/helix/ConfigAccessor.java]: `IllegalStateException` floated from HttpRoutingDataReader should also be caught.

Comment (narendly) [reply helix-core/src/main/java/org/apache/helix/ConfigAccessor.java]: Good catch! Got those sharp eyes on you :)

**Comment by narendly** [comment]
This PR is ready to be merged, approved by @pkuwm 

----

This diff instruments ConfigAccessor's constructors to make it realm-aware. If ConfigAccessor is unable to start on multi-realm mode, then it falls back to starting on single-realm mode.

**Pull request merged**